azurerm/internal/services/policyinsights/remediations.go

```diff
@@ -0,0 +1,57 @@
+package policyinsights
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/services/policyinsights/mgmt/2019-10-01/policyinsights"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/policyinsights/parse"
+)
+
+// RemediationCreateUpdateAtScope is a wrapper of the 4 CreateOrUpdate functions on RemediationsClient, combining them into one to simplify code.
+func RemediationCreateUpdateAtScope(ctx context.Context, client *policyinsights.RemediationsClient, name string, scope parse.RemediationScopeId, parameters policyinsights.Remediation) (policyinsights.Remediation, error) {
+	switch scope.Type {
+	case parse.AtSubscription:
+		return client.CreateOrUpdateAtSubscription(ctx, scope.SubscriptionId, name, parameters)
+	case parse.AtResourceGroup:
+		return client.CreateOrUpdateAtResourceGroup(ctx, scope.SubscriptionId, scope.ResourceGroup, name, parameters)
+	case parse.AtResource:
+		return client.CreateOrUpdateAtResource(ctx, scope.ScopeId, name, parameters)
+	case parse.AtManagementGroup:
+		return client.CreateOrUpdateAtManagementGroup(ctx, scope.ManagementGroupId, name, parameters)
+	default:
+		return policyinsights.Remediation{}, fmt.Errorf("Error creating Policy Remediation: Invalid scope type %q", scope.Type)
+	}
+}
```

Comment:
Given this will only be called in one resource do we need to have a function for it?

Reply:
Sure, i will move these functions to the corresponding CRUD functions

Reply:
Since the GetAtScope function and DeleteAtScope function are called multiple times, I will keep them, but move them to the resource file.